### PR TITLE
Give the text input fields a retouch

### PR DESCRIPTION
**Pull request by ivarnakken**
# Description

They look more modern, and conform better with our styling elsewhere.

A `prefix` prop has been added, which takes in the name of an icon and turns it into a prefix.
A `removeBorder` props has also been added, and is used in the registration table (see image below).
This PR also fixes a bug where the disabled text fields on dark mode did not look disabled.

A big problem with the input fields is that the width of them are `100%` of the content. Changing this requires a lot of work, and is therefore separated into a different PR, but it'll come. Also, a different PR will tackle `react-select`, and make the styling of that similar to this.

# Result

**Before**
image

**After**
image

---

**Before**
image

**After**
image

# Testing

- [x] I have thoroughly tested my changes.

All types of text fields were checked for.

### CODE REVIEW

**Comment by mattiastofte** [comment]
Looks good! It's nice to have a prop for removing the border, as it looks really clean in the right places. Could we get some screenshots showcasing exactly what the prefix prop does?

**Review by LudvigHz — APPROVED** [review]

**Review by norbye — APPROVED** [review]

File: app/components/Form/TextInput.tsx [hunk 1]
```diff
@@ -1,40 +1,70 @@
...
+}: Props) => {
+  /* New ref is made because text inputs that are not Fields are not given a ref */
+  const newInputRef = useRef<HTMLInputElement>(inputRef);
+
   return (
-    <span className={cx(suffix && styles.inputGroup)}>
+    <Flex
+      alignItems="center"
+      gap={10}
+      style={{ border: removeBorder && 'none' }}
```
Comment (Arashfa0301) [inline app/components/Form/TextInput.tsx:40]: Isn't inline css a bad idea ??

Comment (ivarnakken) [reply app/components/Form/TextInput.tsx:40]: Not in this case, as it's dynamic and directly related to the prop.

File: app/components/Form/TextInput.css [hunk 2]
```diff
@@ -1,54 +1,64 @@
 .input {
-  padding: 10px;
-  font-size: 16px;
-  border: 1px solid var(--border-gray);
-  border-radius: 3px;
-  width: 100%;
+  height: 38px;
```
Comment (eikhr) [inline app/components/Form/TextInput.css:2]: This is way too small for textareas! Breaks the company interest form😠

**Pull request merged**